### PR TITLE
GH-226 Rebuild state when stronger chain found - case only genesis in common

### PR DESCRIPTION
**Pull request by lucafavatella**
In scope of GH-226.
Supersedes PR #255.

### CODE REVIEW

File: apps/aecore/src/aec_chain.erl [hunk 1]
```diff
@@ -1169,6 +1172,7 @@ do_force_insert_headers_internal_1(
             {ok, NewTopBlock} =
                 do_find_highest_block_from_header_hash(NewTopHeaderHash,
                                                        NewHeadersDb, NewBlocksDb),
+            %% TODO Trigger check chain for successor.
```
Comment (lucafavatella) [inline apps/aecore/src/aec_chain.erl]: I need to code unit test that fails without this line implemented, then I need to code (or reuse) function to put here.

Comment (lucafavatella) [reply apps/aecore/src/aec_chain.erl]: I plan to open new PR (depending on this PR) for addressing this point.

File: apps/aecore/src/aec_chain.erl [hunk 2]
```diff
@@ -1159,6 +1159,9 @@ do_force_insert_headers_internal_1(
                   end,
                   HeadersDb,
                   HeaderChainWithWork),
+            {ok, _} = aec_state:force_trees(
```
Comment (sennui) [inline apps/aecore/src/aec_chain.erl]: This is resetting trees, only

File: apps/aecore/src/aec_chain.erl (hunk 1 above)
Comment (sennui) [reply apps/aecore/src/aec_chain.erl]: OK, and this is the second half of work

File: apps/aecore/src/aec_chain.erl (hunk 2 above)
Comment (lucafavatella) [reply apps/aecore/src/aec_chain.erl]: Correct.

**Comment by lucafavatella** [comment]
Closing this as #259 now open - more complete as it caters for case missing in this PR.

**Pull request closed without merging**